### PR TITLE
issue #1402 BetterInternetAddress test

### LINKED ISSUE #1402: Add tests for BetterInternetAddress
It would be nice to have tests for BetterInternetAddress(https://en.wikipedia.org/wiki/Email_address#Internationalization_examples)

### Valid email addresses

- [x] simple@example.com
- [x] very.common@example.com
- [x] disposable.style.email.with+symbol@example.com
- [x] other.email-with-hyphen@example.com
- [x] fully-qualified-domain@example.com
- [x] user.name+tag+sorting@example.com (may go to user.name@example.com inbox depending on mail server)
- [x] x@example.com (one-letter local-part)
- [x] example-indeed@strange-example.com
- [x] test/test@test.com (slashes are a printable character, and allowed)
- [ ] admin@mailserver1 (local domain name with no TLD, although ICANN highly discourages dotless email addresses[10])
- [x] example@s.example (see the List of Internet top-level domains)
- [ ] " "@example.org (space between the quotes)
- [x] "john..doe"@example.org (quoted double dot)
- [x] mailhost!username@example.org (bangified host route used for uucp mailers)
- [x] user%example.com@example.org (% escaped mail route to user@example.com via example.org)
- [x] user-@example.org (local part ending with non-alphanumeric character from the list of allowed printable characters)

### Invalid email addresses

- [x] Abc.example.com (no @ character)
- [x] A@b@c@example.com (only one @ is allowed outside quotation marks)
- [x] a"b(c)d,e:f;g<h>i[j\k]l@example.com (none of the special characters in this local-part are allowed outside quotation marks)
- [x] just"not"right@example.com (quoted strings must be dot separated or the only element making up the local-part)
- [x] this is"not\allowed@example.com (spaces, quotes, and backslashes may only exist when within quoted strings and preceded by a backslash)
- [x] this\ still\"not\\allowed@example.com (even if escaped (preceded by a backslash), spaces, quotes, and backslashes must still be contained by quotes)
- [x] 1234567890123456789012345678901234567890123456789012345678901234+x@example.com (local-part is longer than 64 characters)
- [x] i_like_underscore@but_its_not_allowed_in_this_part.example.com (Underscore is not allowed in domain part)
- [x] QA[icon]CHOCOLATE[icon]@test.com (icon characters)

Also, we should support localhost for local testing
- [x] simple@localhost
- [x] "Personal Name <simple@localhost>"

### PR DESCRIPTION
**Pull request by IvanPizhenko**
This PR adds unit tests for the class `BetterInternetAddress` and some fixes.

close #1402

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/main/java/com/flowcrypt/email/util/BetterEmailAddress.kt [hunk 1]
```diff
@@ -29,16 +30,20 @@ class BetterInternetAddress(str: String, verifySpecialCharacters: Boolean = true
     private val containsSpecialCharacterRegex = ".*[()<>\\[\\]:;@\\\\,.\"].*".toRegex()
     // double quotes at ends only
     private val doubleQuotedTextRegex = "\"[^\"]*\"".toRegex()
-    private val validLocalhostEmailRegex = Regex("([a-zA-z])([a-zA-z0-9])+@localhost")
+    private val validLocalhostEmailRegex = Regex("([a-zA-z])([a-zA-z0-9])*@localhost")
+    private const val maxLocalPartLength = 64
```
Comment (DenBond7) [inline FlowCrypt/src/main/java/com/flowcrypt/email/util/BetterEmailAddress.kt:34]: Please make it UPPER_CASE

File: FlowCrypt/src/main/java/com/flowcrypt/email/util/BetterEmailAddress.kt [hunk 2]
```diff
@@ -12,9 +12,10 @@ import com.flowcrypt.email.extensions.kotlin.isValidEmail
 class BetterInternetAddress(str: String, verifySpecialCharacters: Boolean = true) {
 
   companion object {
-    const val alphanum = "\\p{L}\\u0900-\\u097F0-9"
-    const val validEmail = "(?:[${alphanum}!#\$%&'*+/=?^_`{|}~-]+(?:\\.[${alphanum}!#\$%&'*+/=?^" +
-        "_`{|}~-]+)*|\"(?:[\\x01-\\x08" +
+    private const val alphanum = "\\p{L}\\u0900-\\u097F0-9"
```
Comment (DenBond7) [inline FlowCrypt/src/main/java/com/flowcrypt/email/util/BetterEmailAddress.kt:15]: Please make it UPPER_CASE

File: FlowCrypt/src/main/java/com/flowcrypt/email/util/BetterEmailAddress.kt [hunk 3]
```diff
@@ -12,9 +12,10 @@ import com.flowcrypt.email.extensions.kotlin.isValidEmail
...
+    // This regex is based on the one provided here: https://emailregex.com
+    private const val validEmail = "(?:[${alphanum}!#\$%&'*+/=?^_`{|}~-]+(?:\\.[${alphanum}!#" +
```
Comment (DenBond7) [inline FlowCrypt/src/main/java/com/flowcrypt/email/util/BetterEmailAddress.kt:17]: Please make it UPPER_CASE

**Review by DenBond7 — CHANGES_REQUESTED** [review]
Please rename constants. Anyway, looks good

**Comment by DenBond7** [comment]
@IvanPizhenko It seems that one test fails

image

**Comment by DenBond7** [comment]
I've checked the tests after the last changes. They were failed.

image

@IvanPizhenko  Is there expected behavior?

**Comment by IvanPizhenko** [comment]
it needs to be fixed

**Review by DenBond7 — APPROVED** [review]

**Comment by DenBond7** [comment]
It seems we have some problems with the execution of the tests. I'm working on it.

**Comment by DenBond7** [comment]
@IvanPizhenko After merge #1415, please update the current branch and retry

**Pull request merged**